shared/util.go

```diff
@@ -1358,3 +1358,35 @@ func TargetDetect(target string) (targetNode string, targetGroup string) {
 
 	return
 }
+
+// ApplyDeviceOverrides handles the logic for applying device overrides.
+// Receives the profile and local devices and the device overrides.
+// Returns the resulting list of devices.
+func ApplyDeviceOverrides(localDevices map[string]map[string]string, profileDevices map[string]map[string]string, deviceOverrides map[string]map[string]string) (map[string]map[string]string, error) {
+	devicesMap := localDevices
```

Comment:
Is there a specific reason to reassign the map here? Wouldn't the modification on `devicesMap` still be propagated to `localDevices` as its using the same map underneath?

Reply:
I thought it made sense because now it represents the final list of devices and not the initial local devices but in second thought I agree that it is unnecessary so I removed it